src/apify_client/clients/base/actor_job_base_client.py

```diff
@@ -1,13 +1,40 @@
-from typing import Optional
+import math
+from datetime import datetime
+from typing import Dict, Optional
 
+from ..._consts import TERMINAL_ACTOR_JOB_STATUSES, ActorJobStatus
+from ..._utils import _parse_date_fields, _pluck_data
 from .resource_client import ResourceClient
 
-# from ..._utils import _parse_date_fields, _pluck_data
-
 
 class ActorJobBaseClient(ResourceClient):
     """Base sub-client class for actor runs and actor builds."""
 
-    def _wait_for_finish(self, wait_secs: Optional[int] = None) -> None:
-        # TODO implement when implementing run and build endpoints
-        pass
+    def _wait_for_finish(self, wait_secs: Optional[int] = None) -> Dict:
+        started_at = datetime.now()
+        should_repeat = True
+        job: Optional[Dict] = None
+        seconds_elapsed = 0
+
+        while(should_repeat):
+            wait_for_finish = 9999
+            if wait_secs is not None:
+                wait_for_finish = wait_secs - seconds_elapsed
+
+            response = self.http_client.call(
+                url=self.url,
```

Comment:
Let's use `url=self._url()` here for consistency with other resource clients, if we ever decide to change the URL generating logic we can change it in just that one function.